samtranslator/model/api/http_api_generator.py

```diff
@@ -87,6 +98,137 @@ def _construct_http_api(self):
+            basepaths = [self.domain.get("BasePath")]
+        elif self.domain.get("BasePath") and isinstance(self.domain.get("BasePath"), list):
+            basepaths = self.domain.get("BasePath")
+        else:
+            basepaths = None
+
+        basepath_resource_list = []
+
+        if basepaths is None:
+            basepath_mapping = ApiGatewayV2ApiMapping(
+                self.logical_id + "ApiMapping", attributes=self.passthrough_resource_attributes
+            )
+            basepath_mapping.DomainName = ref(self.domain.get("ApiDomainName"))
+            basepath_mapping.ApiId = ref(http_api.logical_id)
+            basepath_mapping.Stage = ref(http_api.logical_id + ".Stage")
+            basepath_resource_list.extend([basepath_mapping])
+        else:
+            for path in basepaths:
+                invalid_regex = r"[^0-9a-zA-Z\/\-\_]+"
+                if re.search(invalid_regex, path) is not None:
+                    raise InvalidResourceException(self.logical_id, "Invalid Basepath name provided.")
+                m = re.search(r"[a-z0-9]+[\-\_]?[a-z0-9]+", path)
+                path = m.string[m.start(0) : m.end(0)]
+                if path is None:
+                    raise InvalidResourceException(self.logical_id, "Invalid Basepath name provided")
+                logical_id = "{}{}{}".format(self.logical_id, re.sub(r"[\-\_]+", "", path), "ApiMapping")
+                basepath_mapping = ApiGatewayV2ApiMapping(logical_id, attributes=self.passthrough_resource_attributes)
+                basepath_mapping.DomainName = ref(self.domain.get("ApiDomainName"))
+                basepath_mapping.ApiId = ref(http_api.logical_id)
+                basepath_mapping.Stage = ref(http_api.logical_id + ".Stage")
+                basepath_mapping.ApiMappingKey = path
+                basepath_resource_list.extend([basepath_mapping])
+
+        # Create the Route53 RecordSetGroup resource
+        record_set_group = None
+        if self.domain.get("Route53") is not None:
+            route53 = self.domain.get("Route53")
+            if route53.get("HostedZoneId") is None and route53.get("HostedZoneName") is None:
+                raise InvalidResourceException(
+                    self.logical_id,
+                    "HostedZoneId or HostedZoneName is required to enable Route53 support on Custom Domains.",
+                )
+            logical_id = logical_id_generator.LogicalIdGenerator(
+                "", route53.get("HostedZoneId") or route53.get("HostedZoneName")
+            ).gen()
+            record_set_group = Route53RecordSetGroup(
+                "RecordSetGroup" + logical_id, attributes=self.passthrough_resource_attributes
+            )
+            if "HostedZoneId" in route53:
+                record_set_group.HostedZoneId = route53.get("HostedZoneId")
+            if "HostedZoneName" in route53 and "HostedZoneId" not in route53:
```

Comment:
If `HostedZoneId` and `HostedZoneName` are given and both are in route53 then do we add only `HostedZoneId` to `record_set_group` and ignore `HostedZoneName`?

Reply:
yes, because in [cfn docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-route53-recordset.html#cfn-route53-recordset-hostedzoneid) this is the guideline - "Specify either HostedZoneName or HostedZoneId, but not both."

Reply:
got it. Thank you